pandas/core/indexes/datetimelike.py

```diff
@@ -612,9 +712,11 @@ def index_method(self, *args, **kwargs):
         result = method(self, *args, **kwargs)
 
         # Index.__new__ will choose the appropriate subclass to return
-        result = Index(result)
-        if pin_name:
-            result.name = self.name
+        if box:
+            result = Index(result)
```

Comment:
we should add a name attribute to EA

Reply:
I don't think so. ndarrays and pyarrow arrays don't have names. Names feel like they apply to containers around arrays.